parser/src/main/java/me/vertretungsplan/parser/IphisParser.java

```diff
@@ -0,0 +1,358 @@
+    /**  */
+    private String authToken;
+
+    public IphisParser(SubstitutionScheduleData scheduleData, CookieProvider cookieProvider) {
+        super(scheduleData, cookieProvider);
+        JSONObject data = scheduleData.getData();
+        try {
+            api = "https://" + data.getString(PARAM_URL) + "/remote/vertretungsplan/ssp";
+            kuerzel = data.getString(PARAM_KUERZEL);
+            website = data.getString(PARAM_WEBSITE);
+            jwt_key = data.getString(PARAM_JWT_KEY);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+    }
+
+    public SubstitutionSchedule getSubstitutionSchedule()
+            throws IOException, JSONException, CredentialInvalidException {
+        final SubstitutionSchedule substitutionSchedule = SubstitutionSchedule.fromData(scheduleData);
+
+        grades = getGrades();
+        teachers = getTeachers();
+        final JSONArray changes = getChanges();
+
+        substitutionSchedule.setClasses(getAllClasses());
+        substitutionSchedule.setTeachers(getAllTeachers());
+        substitutionSchedule.setWebsite(website);
+
+        parseIphis(substitutionSchedule, changes, grades, teachers);
+        return substitutionSchedule;
+    }
+
+    /**
+     * Returns authentication key
+     */
+    private String getAuthenticationHeader() throws JSONException, IOException, CredentialInvalidException {
+        final UserPasswordCredential userPasswordCredential = (UserPasswordCredential) credential;
+        final String username = userPasswordCredential.getUsername();
+        final String password = userPasswordCredential.getPassword();
+
+        if (authToken == null) {
+            JSONObject payload = new JSONObject();
+            payload.put("school", kuerzel);
+            payload.put("user", username);
+            payload.put("password", password);
+
+            httpPost(api + "/login", "UTF-8", payload.toString(), ContentType.APPLICATION_JSON);
+            final String token = httpPost(api + "/login", "UTF-8", payload.toString(), ContentType.APPLICATION_JSON);
+
+            try {
+                final String key = Base64.getEncoder().encodeToString(jwt_key.getBytes());
+                final Claims jwtToken = Jwts.parser().setSigningKey(key)
+                        .parseClaimsJws(token).getBody();
+                assert jwtToken.getSubject().equals("vertretungsplan.me");
+
+                authToken = token;
+                website = jwtToken.getIssuer();
+                if (username.equals("lehrer")) {
+                    getSubstitutionSchedule().setType(SubstitutionSchedule.Type.TEACHER);
+                } else {
+                    getSubstitutionSchedule().setType(SubstitutionSchedule.Type.STUDENT);
+                }
+                getSubstitutionSchedule().setLastChange(LocalDateTime.fromDateFields(jwtToken.getIssuedAt()));
+                getSubstitutionSchedule().setLastChangeString(jwtToken.getIssuedAt().toString());
```

Comment:
Bei anderen Parsern ist `setLastChangeString` ein Fallback falls das Datum in einem unerwarteten Format ist. Wenn das Parsen des Datums wie hier sicher klappt, ist `setLastChangeString` nicht nötig, da wird dann automatisch das Datum zu einem String formatiert.